lib/ii.c

```diff
@@ -11768,14 +11768,18 @@ grn_ii_select_data_check_near_element_intervals(grn_ctx *ctx,
   } else if (data->mode == GRN_OP_ORDERED_NEAR_PHRASE) {
     uint32_t i;
     uint32_t n = data->n_token_infos;
-    if (n > n_max_element_intervals + 1) {
-      n = n_max_element_intervals + 1;
-    }
     int32_t previous_pos = data->token_infos[0]->pos;
-    for (i = 1; i < n; i++) {
+    uint32_t n_checked_elements = 0;
+
+    for (i = 1; i < n && n_checked_elements < n_max_element_intervals; i++) {
```

Comment:
We should compare the first token in the `i-1`th phrase and the first token in the `i`th phrase.
Does this change do it?

Reply:
Sounds great!
I didn't noticed that we have `phrase_id`.

Reply:
Ah, the code is buggy. `n_max_element_intervals + 1` should be `n_max_element_intervals`.

Reply:
I have added a little modification about `n_phrases`, and it works fine!
Thanks!